### PR TITLE
OOP school library: refactor your code

### PR DESCRIPTION
**Pull request by muskan2532150**
-  we refactored the code to follow SOLID principles.
- We followed the YAGNI, KISS, and DRY rules, and special attention was given to the "Single-responsibility Principle".
- The role of the main file is only to start the app and ask the user for the initial choice of options (or to quit).
- Other methods were extracted into separate classes.

### CODE REVIEW

File: create_module.rb [hunk 1]
```diff
@@ -0,0 +1,52 @@
+module CreateElement
+  def create_book
+    print 'Title: '
+    title = gets.chomp
+    print 'Author: '
+    author = gets.chomp
```
Comment (HAMMAS-SALEEM) [inline create_module.rb]: - [ ] You are using print and gets.chomp at a lot of places means that you are repeating the code so your code is not DRY. Kindly make a separate function for taking inputs 👍 to reuse that everywhere you wanna get inputs 😊

Hint: See the function below ⬇️
```
def inputs(show)
  outputs = []
  show.each do |s|
    print s
    outputs.push(gets.chomp)
  end
  outputs
end
```

File: create_module.rb [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    print 'Classroom: '
+    classroom = gets.chomp
+    print 'Name: '
+    name = gets.chomp
+    print 'Has Parent Permission (true/false): '
+    has_permission = gets.chomp
+    student = Student.new(name, age.to_i, has_permission, classroom)
+    @person.push(student) unless @person.include?(student)
+    puts 'Record is created successfully!!'
+  end
```
Comment (HAMMAS-SALEEM) [inline create_module.rb]: - [ ] All your methods should be SOLID 🤔 means they should be responsible for single purpose. But in your code all your methods are having more than one responsibility such as the above method is taking inputs (First Responsibility), creating students (Second Responsibility) 🤔 Kindly make all your methods single responsible 😊

**Review by HAMMAS-SALEEM — CHANGES_REQUESTED** [review]
# Required Changes :recycle:
almost-there

## Hey @muskan2532150, @yasinabdmahmood, @dlwub :wave:

## Highlights :1st_place_medal:

✔️ Descriptive PR Title
✔️ Descriptive PR Summary
✔️ Following Gitflow
✔️ Following some requirements

Good job on implementing some of the requirements, there are some issues that you still need to work on to go to the next project but you are almost there! :blush:

# Optional

_All the changes with optional tags are not crucial enough to prevent you from getting the approval but we highly recommend to students that they should take these changes into consideration._

Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

---------------------------------------

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by omar-muhamad — APPROVED** [review]
# Approved ✔
### Hi Team,

Great job, Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

thumbs-up-agree


## (Highlights) Good Points: :+1:
- You documented your work professionally. ✔
- You used the correct workflow.  ✔
- All linters checks are ok. ✔

## (Changes Required) Aspects to improve: ♻️

- N/A

## [Optional] suggestions:

- N/A

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**